tests/testthat/test-independent-testing-tlf_ae_summary.R

```diff
@@ -0,0 +1,25 @@
+library(metalite)
+library(metalite.ae)
+
+tempdir()
+meta <- meta_ae_dummy()
+outdata <- prepare_ae_summary(meta,
+                              population = "apat",
+                              observation = "wk12",
+                              parameter = "any;rel;ser")
+tbl <- outdata |>
+  format_ae_summary()
+
+tbl |>
+  tlf_ae_summary(
+    source = "Source:  [CDISCpilot: adam-adsl; adae]",
+    path_outdata = paste0(tempdir(),"/aesummary2.rdata"),
```

Comment:
Hi @howardbaek and @ginnaram13 , could you please not use `tempdir()`? It will fail the CMD check. Please save the outdata and outtable into a sub-folder under the testthat folder. And you can call this subfolder as fixtures. Then I guess the following works:
```
path_outdata = "./fixtures/aesummary2.rdata"
path_outtable = "./fixtures/aesummary2.rtf"
```

Reply:
```
library(metalite)
library(metalite.ae)

meta <- meta_ae_dummy()
outdata <- prepare_ae_summary(meta,
                              population = "apat",
                              observation = "wk12",
                              parameter = "any;rel;ser")
tbl <- outdata |>
  format_ae_summary()

test_that("aesummary.rtf file output", {
  expect_snapshot_file(tbl |>
                         tlf_ae_summary(
                           source = "Source:  [CDISCpilot: adam-adsl; adae]",
                           path_outtable = "./tests/testthat/fixtures/aesummary2.rtf"),
                       "./tests/testthat/fixtures/aesummary2.rtf")
})

test_that("aesummary.rdata file output", {
  expect_snapshot_file(tbl |>
                         tlf_ae_summary(
                           source = "Source:  [CDISCpilot: adam-adsl; adae]",
                           path_outdata = "./tests/testthat/fixtures/aesummary2.rdata"),
                       "./tests/testthat/fixtures/aesummary2.rdata")
})
```

Above is my updated code, but I don't understand why I keep on failing tests..

Reply:
The first component in `expect_snapshot_file` is wrong. `tbl |>ntlf_ae_summary( source = "Source:  [CDISCpilot: adam-adsl; adae]", path_outtable = "./tests/testthat/fixtures/aesummary2.rtf")` automatically generate rtf and data and save it under the `path_outdata`. It doesn't say anything about the generated rtf. 
For your `expect_snapshot_file` , your objective is to compare 2 rtf. So the  `tbl |>ntlf_ae_summary( source = "Source:  [CDISCpilot: adam-adsl; adae]", path_outtable = "./tests/testthat/fixtures/aesummary2.rtf")`  should be a rtf object.

Reply:
I have changed the code for snapshot testing and created an example. It should work fine now. Please try it and let me know if there is any issue. Also, I found `Rplots.pdf` in the `testthat` folder, do we need it?

Reply:
Can you tell me why `tempdir()` will fail CMD check? Is there some documentation about this?

Reply:
I was wondering about @BrianLang 's question too. 

@wangben718 I tried your changes to the code and it works great. Also, I don't know where `Rplots.pdf` came from, but I think we can delete it? I tried opening the pdf file, but I got an error. 

One last thing: I'm a bit confused as to why Ben's code works. I can see why we use `file.path()` and `testthat::test_path()`, but I don't get why we just need the `file.path()` inside `expect_snapshot_file()`.  If you look at the Examples in the Help page for `expect_snapshot_file()`, there is this line of code: `expect_snapshot_file(save_png(hist(mtcars$mpg)), "plot.png")`. 

I thought we needed to follow this format: `expect_snapshot_file(code_that_returns_path_to_file(), file name)`, but it seems like `expect_snapshot_file(file name)` passes the tests.

Reply:
`tempdir()` should work in most cases. The only case I could imaging is that github action may not allow user to create new folders. But I am not 100% sure. 

Pass snapshot test can be challenge in a remote server due to different OS. I would suggest to not perform snapshot check for an `Rdata` and `pdf` file. For PDF file, it will contain timestamp in the file that will not pass snapshot. I guess it could also be the reason for `Rdata`.